Source/Core/VideoCommon/UberShaderCommon.cpp

```diff
@@ -94,8 +94,8 @@ void WriteVertexLighting(ShaderCode& out, APIType api_type, const char* world_po
                          const char* out_color_1_var)
 {
   out.Write("// Lighting\n");
-  out.Write("%sfor (uint chan = 0u; chan < xfmem_numColorChans; chan++) {\n",
-            api_type == APIType::D3D ? "[loop] " : "");
+  out.Write("%sfor (uint chan = 0u; chan < %uu; chan++) {\n",
+            api_type == APIType::D3D ? "[loop] " : "", static_cast<u32>(NUM_XF_COLOR_CHANNELS));
```

Comment:
Why do you cast to unsigned here? %du should be fine as well.

Reply:
NUM_XF_COLOR_CHANNELS is size_t. I've never checked the spec, so I'm not sure if it's okay to print a size_t as %u on 64-bit?

Reply:
Wouldn't it be sufficient to use %zu in that case?

Reply:
Yep, I've changed it to %zu. TIL about this specifier.